### DISCUSSION TITLE
is there a way to run cron job for every 10 seconds or 30 seconds

### DISCUSSION THREAD

**Discussion by nikhil-varma-mudunuri** [pocketbase/pocketbase#3535] [Q&A] [answered]
same as title

**Comment by ganigeorgiev** [comment, marked as answer]
If you mean for the JS `pb_hooks`, then No (usually cron jobs have 1 minute granularity).

If you use Go, then you can specify a custom ticker interval for the specific cron instance with [`cron.SetInterval`](https://pkg.go.dev/github.com/pocketbase/pocketbase@v0.19.0/tools/cron#Cron.SetInterval) but even with this the cron expression format doesn't have a seconds notation.
For more granular tickers you can create your own [`time.Ticker`](https://pkg.go.dev/time#NewTicker).

**Comment by romechenko** [comment]
I'm trying out six cron jobs that are started with a delay of [0, 10, 20, 30, 40, 50] seconds.

**Reply by romechenko** [reply]
finding it hard so far because of lack of setTimeout
https://pocketbase.io/docs/js-overview#engine-limitations

**Reply by ganigeorgiev** [reply]
A "cron job" by spec doesn't have seconds granularity.

Could you elaborate more on your use case and why do you need it?

**Reply by romechenko** [reply]
Thanks for the reply
I want to process a task from a queue every 10 seconds

**Reply by romechenko** [reply]
I see this question and you had replied with a suggestion. I was hoping to see an example of how someone had done that.
https://github.com/pocketbase/pocketbase/discussions/2101

**Reply by romechenko** [reply]
It would go a good part of the way if the suggested solution used js instead of go

**Reply by ganigeorgiev** [reply]
That doesn't really help understanding what is the purpose of it, but in any case as mentioned above in https://github.com/pocketbase/pocketbase/discussions/3535#discussioncomment-7302280, with the JS `pb_hooks` at the moment is not possible to execute repetitive tasks < 1min granularity.

It is possible to enable an event loop for the JS VM which will allow access to `setTimeout`/`setInterval` but this could have some unintended side-effects as it will require blocking/keeping alive the goja runtimes for longer and for now I want to avoid it.

**Reply by romechenko** [reply]
That makes sense.
A bit more context about my problem is that I have an expensive process that I want to run if a record in a collection has been changed but I want to allow 10 seconds or so incase more changes come in. I also expect that there might be multiple records that want to trigger this expensive process so I want to add it to a queue to be sure I'm not processing more than n at a time.

**Comment by AxelRothe** [comment]
@ImJohnDoe I use a self hosted service like windmill to run my cron jobs, by logging in as an admin via the JS Client. That way I can monitor the jobs easily. https://www.windmill.dev

**Reply by romechenko** [reply]
That’s a great idea. Simple and straightforward. Windmill looks solid. Thanks for the comment!
